src/libraries/Microsoft.Extensions.Options/gen/DiagDescriptors.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using System;
+
+namespace Microsoft.Extensions.Options.Generators
+{
+    internal sealed class DiagDescriptors : DiagDescriptorsBase
+    {
+        private const string Category = "Microsoft.Extensions.Options.SourceGeneration";
+
+        public static DiagnosticDescriptor CantUseWithGenericTypes { get; } = Make(
+            id: "SYSLIB1201",
```

Comment:
Register these diagnostics in `list-of-diagnostics.md`?https://github.com/dotnet/runtime/blob/f450f492ca266e9c5c78266dcd4781ea8907cf48/docs/project/list-of-diagnostics.md?plain=1#L219C3-L237